astacus/coordinator/plugins/clickhouse/object_storage.py

```diff
@@ -91,12 +91,21 @@ def _copy_items_between(
         target_storage: BaseTransfer[Any],
         stats: StatsClient | None,
     ) -> None:
+        total_keys = len(keys)
         match self.copy_config:
             case DirectCopyConfig():
-                logger.info("Copying the keys using the cloud APIs")
-                target_storage.copy_files_from(source=source_storage, keys=keys)
+                logger.info("Copying %r keys using the cloud APIs", total_keys)
+                target_storage.copy_files_from(
+                    source=source_storage,
+                    keys=keys,
+                    progress_fn=lambda completed_files, total_files: _handle_copy_progress_fn(
+                        completed_files, total_files, self.copy_config.method, stats
+                    )
+                    if stats
+                    else None,
+                )
             case LocalCopyConfig():
-                logger.info("Copying the keys by downloading from source/uploading to target")
+                logger.info("Copying %r keys by downloading from source/uploading to target", total_keys)
```

Comment:
Since metrics are not a reliable data source, we add in addition the initial amount of keys for debugging purpose.

Reply:
That's cool! The keyword should be `%i` instead of `%r`